destination/clickhouse.go

```diff
@@ -14,168 +15,157 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
-type WriteBatchOpType string
-
-const (
-	Replace    WriteBatchOpType = "Replace"
-	Update     WriteBatchOpType = "Update"
-	SoftDelete WriteBatchOpType = "SoftDelete"
-)
-
 // ClickHouseConnection
 // TODO: on premise cluster setup for DDL
 type ClickHouseConnection struct {
-	Database string
-	ctx      context.Context
+	ctx context.Context
 	driver.Conn
 }
 
 func GetClickHouseConnection(ctx context.Context, configuration map[string]string) (*ClickHouseConnection, error) {
-	return RetryNetErrorWithData(func() (*ClickHouseConnection, error) {
-		hostname := fmt.Sprintf("%s:%s",
-			GetWithDefault(configuration, "hostname", "localhost"),
-			GetWithDefault(configuration, "port", "9000"))
-		username := GetWithDefault(configuration, "username", "default")
-		password := GetWithDefault(configuration, "password", "")
-		database := GetWithDefault(configuration, "database", "default")
-		options := &clickhouse.Options{
-			Addr: []string{hostname},
-			Auth: clickhouse.Auth{
-				Username: username,
-				Password: password,
-				Database: database,
-			},
-			Protocol: clickhouse.Native,
-			Settings: clickhouse.Settings{
-				"allow_experimental_object_type": 1,
-				"date_time_input_format":         "best_effort",
-			},
-			MaxOpenConns: int(*maxOpenConnections),
-			MaxIdleConns: int(*maxIdleConnections),
-			ClientInfo: clickhouse.ClientInfo{
-				Products: []struct {
-					Name    string
-					Version string
-				}{
-					{Name: "fivetran-destination", Version: Version},
-				},
+	hostname := fmt.Sprintf("%s:%s",
```

Comment:
do you want to implement a persistent connection now or decided to postpone?

Reply:
let's postpone